File: app/src/main/java/de/tum/in/tumcampusapp/cards/generic/StickyCard.java
package de.tum.in.tumcampusapp.cards.generic;

import android.content.Context;
import android.content.SharedPreferences;

public abstract class StickyCard extends Card {

    public StickyCard(int cardType, Context context) {
        super(cardType, context);
    }

    @Override
    public final boolean isDismissible() {
        return false;
    }

    @Override
    protected final void discard(SharedPreferences.Editor editor) {
        // Sticky cards can't be dismissed
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/card/StickyCard.kt
package de.tum.`in`.tumcampusapp.component.ui.overview.card

import android.content.Context
import android.content.SharedPreferences
import de.tum.`in`.tumcampusapp.component.ui.overview.CardManager

abstract class StickyCard(cardType: CardManager.CardTypes, context: Context) : Card(cardType, context) {

    override val isDismissible: Boolean
        get() = false

    override fun discard(editor: SharedPreferences.Editor) {
        // Sticky cards can't be dismissed
    }
}
